internal/services/hybridcompute/hybrid_compute_machine_data_source_test.go

```diff
@@ -0,0 +1,180 @@
+package hybridcompute_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+)
+
+type HybridComputeMachineDataSource struct{}
+
+func TestAccHybridComputeMachine_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_hybrid_compute_machine", "test")
+	d := HybridComputeMachineDataSource{}
+	clientSecret := os.Getenv("ARM_CLIENT_SECRET")
+	randomUUID, _ := uuid.GenerateUUID()
+	data.DataSourceTest(t, []acceptance.TestStep{
+		{
+			Config: d.basic(data, clientSecret, randomUUID),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("agent_configuration.#").HasValue("1"),
```

Comment:
Are there any other checks we can reliably add here, given the complexity of this data source?